src/ChunkedDecoder.php

```diff
@@ -0,0 +1,153 @@
+    {
+        Util::pipe($this, $dest, $options);
+
+        return $dest;
+    }
+
+    public function close()
+    {
+        if ($this->closed) {
+            return;
+        }
+
+        $this->buffer = '';
+
+        $this->closed = true;
+
+        $this->input->close();
+
+        $this->emit('close');
+        $this->removeAllListeners();
+    }
+
+    /** @internal */
+    public function handleEnd()
+    {
+        if (!$this->closed) {
+            $this->handleError(new \Exception('Unexpected `end` event'));
+        }
+    }
+
+    /** @internal */
+    public function handleError(\Exception $e)
+    {
+        $this->emit('error', array($e));
+        $this->input->removeListener('data', array($this, 'handleData'));
+        $this->close();
+    }
+
+    /** @internal */
+    public function handleData($data)
+    {
+        $this->buffer .= $data;
+
+        while ($this->buffer !== '') {
+            if (!$this->headerCompleted) {
+                $positionClrf = strpos($this->buffer, static::CRLF);
+
+                if ($positionClrf === false) {
+                    // Header shouldn't be bigger than 1024 bytes
+                    if (strlen($this->buffer) > 1024) {
+                        $this->handleError(new \Exception('Chunk size inclusive extension bigger than 1024 bytes'));
+                    }
+                    return;
+                }
+
+                $header = strtolower((string)substr($this->buffer, 0, $positionClrf));
+                $hexValue = $header;
+                $start = strlen($hexValue) + 2;
+
+                if (strpos($header, ';') !== false) {
+                    $array = explode(';', $header);
+                    $hexValue = $array[0];
+                    $start = strlen($header) + 2;
+                }
+
+                if (dechex(hexdec($hexValue)) !== $hexValue || hexdec($hexValue) > 2147483647) {
+                    $this->handleError(new \Exception($hexValue . ' is not a hexadecimal number or too big'));
+                    return;
+                }
+
+                $this->chunkSize = hexdec($hexValue);
+                $this->buffer = (string)substr($this->buffer, $start);
+                $this->headerCompleted = true;
+                if ($this->buffer === '') {
+                    return;
+                }
+            }
+
+            $chunk = (string)substr($this->buffer, 0, $this->chunkSize - $this->transferredSize);
+            if ($chunk !== '') {
+                $this->transferredSize += strlen($chunk);
+                if ($this->transferredSize > $this->chunkSize) {
+                    $this->handleError(new \Exception('The chunk is bigger than expected'));
+                    return;
+                }
+
+                $this->emit('data', array($chunk));
+                $this->buffer = (string)substr($this->buffer, strlen($chunk));
+            }
+
+            if (strpos($this->buffer, static::CRLF) !== false) {
```

Comment:
Is this resolved? This still checks for *any* position afaict? `4\r\ntestWOOT\r\n`